[babycare_product_brand][IMP] make strings in product.py unique because otherwise there are duplicates in the advanced search

wytzejan opened the pull request:
Each field in the product attribute manager is made unique.

**Question**
The purpose of this change is to have unique names in the advanced search of the product variant model. (E.g.: in case we would like to mass-edit a field we need to know to which product group this field belongs to.) Disadvantage of this change is that also the names of the fields on the product form are changed.
Is there a way to have different names for the fields in the product form and the advanced search filtering? (E.g.: for the field product_buggies_agecategory_id we would like to have "Age Category (Buggies)" displayed in the advanced filtering but "Age Category" on the product form.)

New translations has been added. For this I created a new nl.po file and replaced this file with the old file in the i18n-folder. Is this the correct way to update translations?

wytzejan committed:
[babycare_product_brand][IMP] make strings in product.py unique because otherwise there are duplicates in the advanced search

StefanRijnhart commented:
This will set the new translations upon a fresh installation, but it will not update the translations when the module was installed before. In Odoo, there is the option to overwrite all the translations in the database with the translations from all the modules, but it says in my notes about your setups that you have customized some translations in the interface and this process would lead to loss of custom translations. So in order for the translations to load, you would need to bump the module version and add a pre-migration script that removes the existing translations for this module only:

```
def migrate(cr):
    cr.execute("DELETE FROM ir_translation WHERE module = 'babycare_product_brand';")
```

You can find an example of a pre-migration script here: https://github.com/babycarenl/babycare-odoo/blob/8.0/bbc_sale/migrations/8.0.1.2/pre-migrate.py

wytzejan committed:
[babycare_product_brand][FIX] pre-migrate script added for updating new translation file

wytzejan commented:
@StefanRijnhart Thanks for your helpful comment. I added a pre-migration script for updating the translations.

**Can you share your thoughts about the following. Please also check the task in Odoo; I've added screenshots in the task.**
The purpose of this change is to have unique names in the advanced search of the product variant model. (E.g.: in case we would like to mass-edit a field we need to know to which product group this field belongs to.) Disadvantage of this change is that also the names of the fields on the product form are changed.
Is there a way to have different names for the fields in the product form and the advanced search filtering? (E.g.: for the field product_buggies_agecategory_id we would like to have "Age Category (Buggies)" displayed in the advanced filtering but "Age Category" on the product form.)

StefanRijnhart commented:
If you just want to adjust the display name, you can override the name_get function. It means that the new names are not searchable as such.

BTW for easy task recognition, please add task references in your branch names and commits. (I create branches named 'tasks/1234/description', and my commits usually have the #1234 task number in them.

StefanRijnhart commented:
Sorry, I misunderstood. If you want to show a different label on a field in the product form, you can use the string= attribute in the xml field definition on the form.

wytzejan committed:
[babycare_product_brand][FIX] added string names on different fields of product attribute manager

wytzejan commented:
@StefanRijnhart thanks for the solution. It was a quite easy fix.
Regarding your note about task recognition: you're absolutely right. I will work on that ;-)
